### PR TITLE
Improved training configuration

### PR DESCRIPTION
**Pull request by l45k**
This change introduce more flexibility in how a job is send from the scheduler to the worker. It introduces DiLoCoConfigurations that hold configuration options for things like optimizer, epoch, lr-scheduler, etc. Furthermore, it also changes the training code to utilize the introduces configuration options.

### CODE REVIEW

File: crates/messages/src/lib.rs [hunk 1]
```diff
@@ -324,23 +381,51 @@ pub enum Executor {
         model: Fetch,
         data: Fetch,
         updates: Receive,
-        results: Send,
+        results: Box<Send>,
+        config: Box<DiLoCoConfig>,
```
[the remark this answers is not in the document]
Comment (l45k) [reply crates/messages/src/lib.rs:385]: Yes, thats the only reason.

File: crates/messages/src/lib.rs [hunk 2]
```diff
@@ -324,23 +381,51 @@ pub enum Executor {
...
     },
     #[serde(rename = "parameter-server")]
     ParameterServer { updates: Receive, results: Send },
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(tag = "type")]
```
[the remark this answers is not in the document]
Comment (l45k) [reply crates/messages/src/lib.rs:392]: That's a cool feature. Is there a reason why you prefere "kebab-case" over "snake_case"?

**Review by nfnt — APPROVED** [review]
LGTM!
